### PR TITLE
remote_access: set up ssh and openssl before connect

### PR DESCRIPTION
**Pull request by vasekhodina**
Connecting to xen VM now required additional set before connecting to Xen VM. This commit adds the configuration.

Automation of VIRT-18307.

### CODE REVIEW

**Comment by dzhengfy** [comment]
@vasekhodina Please attach your test results for your PR. This is required. Thanks.

File: libvirt/tests/src/remote_access/remote_access.py [hunk 1]
```diff
@@ -305,6 +358,15 @@ def run(test, params, env):
         crypto_policies = process.run("update-crypto-policies --set LEGACY",
                                       ignore_status=False)
 
+    # Another bug check below
+    if driver == "xen":
+        if (ssh_config_path and os.path.exists(ssh_config_path)):
```
[the remark this answers is not in the document]
Comment (vasekhodina) [reply libvirt/tests/src/remote_access/remote_access.py:363]: I have changed the if clause order. Now the path is not created by default but that might change in the future. So I think it's a good idea to check if it's already present.

File: libvirt/tests/src/remote_access/remote_access.py [hunk 2]
```diff
@@ -170,7 +172,54 @@ def change_libvirtconf_on_client(params_to_change_dict):
     return config
 
 
+def change_ssh_conf_on_client(filepath, hostname):
+    """
+    Create a ssh config file for at the filepath specified.
+
+    :param filepath: String, File location to copy the config to
+    :param uri_path: String, hostname of the host in config
```
[the remark this answers is not in the document]
Comment (vasekhodina) [reply libvirt/tests/src/remote_access/remote_access.py:180]: Updated

File: libvirt/tests/src/remote_access/remote_access.py [hunk 3]
```diff
@@ -170,7 +172,54 @@ def change_libvirtconf_on_client(params_to_change_dict):
...
+    """
+    backup_path = filepath + ".bkp"
+    shutil.copy(filepath, backup_path)
+    return backup_path
+
+
 def restore_libvirtconf_on_client(config):
+    """
+    Change libvirt configuration to the one specified in config argument.
+    """
```
[the remark this answers is not in the document]
Comment (vasekhodina) [reply libvirt/tests/src/remote_access/remote_access.py:222]: Updated.

File: libvirt/tests/src/remote_access/remote_access.py [hunk 4]
```diff
@@ -595,3 +657,9 @@ def run(test, params, env):
 
         if polkit_pkla and os.path.isfile(polkit_pkla):
             os.unlink(polkit_pkla)
+
+        if ssh_config_path:
+            shutil.copy(ssh_config_backup_path, ssh_config_path)
```
[the remark this answers is not in the document]
Comment (vasekhodina) [reply libvirt/tests/src/remote_access/remote_access.py:662]: Updated.

[the remark this answers is not in the document]
Comment (dzhengfy) [reply libvirt/tests/src/remote_access/remote_access.py:668]: I did not see where ssh_config_back _up_path file is removed. Could you point me?

[the remark this answers is not in the document]
Comment (dzhengfy) [reply libvirt/tests/src/remote_access/remote_access.py:668]: I did not see where ssh_config_back _up_path file is removed. Could you point me?

File: libvirt/tests/src/remote_access/remote_access.py [hunk 5]
```diff
@@ -595,3 +660,12 @@ def run(test, params, env):
 
         if polkit_pkla and os.path.isfile(polkit_pkla):
             os.unlink(polkit_pkla)
+
+        if ssh_config_path:
+            os.remove(ssh_config_path)
+
+        if ssh_config_backup_path:
```
Comment (dzhengfy) [inline libvirt/tests/src/remote_access/remote_access.py:667]: I did not see where ssh_config_back _up_path file is removed. Could you point me?

File: libvirt/tests/src/remote_access/remote_access.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (dzhengfy) [reply libvirt/tests/src/remote_access/remote_access.py:668]: I did not see your updates for this. I mean the file should be removed.

**Review by dzhengfy — APPROVED** [review]
LGTM

**Comment by dzhengfy** [comment]
@cliping Could you help review this one? Thanks

**Review by cliping — APPROVED** [review]
LGTM

**Pull request merged**